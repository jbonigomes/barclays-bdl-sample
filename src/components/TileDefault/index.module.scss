@import '../../index.scss';

.container {
  width: 270px;
  font-family: 'ExpertSans-Light';

  .content {
    padding: 0 24px 24px 24px;

    & > button {
      margin-top: 24px;
    }
  }

  .cover {
    height: 125px;
    margin-bottom: 20px;
  }

  img {
    width: 100%;
    height: 100%;
    object-fit: fill;
  }

  @media (min-width: $breakpoint) {
    width: 448px;

    .cover {
      height: 224px;
    }
  }
}
