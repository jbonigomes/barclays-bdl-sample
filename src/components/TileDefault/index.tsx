import React from 'react'
import Tile from '../Tile'
import Button from '../Button'
import LeadLink from '../LeadLink'
import styles from './index.module.scss'

type TileDefaultProps = {
  image: string,
  title: string,
  buttonText?: string,
  children: React.ReactNode,
  onTitleClick(e: React.MouseEvent<HTMLElement>): void,
  onButtonClick?: (e: React.MouseEvent<HTMLElement>) => void,
}

const TileDefault: React.SFC<TileDefaultProps> = ({
  image,
  title,
  children,
  buttonText,
  onTitleClick,
  onButtonClick,
}: TileDefaultProps) => (
  <div className={styles.container}>
    <Tile>
      <div className={styles.cover}>
        <img src={image} alt="Tile graphic" />
      </div>
      <div className={styles.content}>
        <LeadLink onClick={onTitleClick} text={title} />
        <div>{children}</div>
        {onButtonClick && buttonText && (
          <Button
            type="primary"
            text={buttonText}
            onClick={onButtonClick}
          />
        )}
      </div>
    </Tile>
  </div>
)

export default TileDefault
