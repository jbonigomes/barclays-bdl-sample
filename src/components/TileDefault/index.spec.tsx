import React from 'react'
import renderer from 'react-test-renderer'

import TileDefault from '.'

test('Match snapshot', () => {
  const props = {
    image: 'image',
    title: 'title',
    onTitleClick: jest.fn(),
  }

  const tree = renderer.create(
    <TileDefault
      image={props.image}
      title={props.title}
      onTitleClick={props.onTitleClick}
    >
      Foo
    </TileDefault>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('Match snapshot with button', () => {
  const props = {
    image: 'image',
    title: 'title',
    buttonText: 'buttonText',
    onTitleClick: jest.fn(),
    onButtonClick: jest.fn(),
  }

  const tree = renderer.create(
    <TileDefault
      image={props.image}
      title={props.title}
      buttonText={props.buttonText}
      onTitleClick={props.onTitleClick}
      onButtonClick={props.onButtonClick}
    >
      Foo
    </TileDefault>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})
