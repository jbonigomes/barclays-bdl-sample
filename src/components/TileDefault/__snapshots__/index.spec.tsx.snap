// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="tile"
  >
    <div
      className="cover"
    >
      <img
        alt="Tile graphic"
        src="image"
      />
    </div>
    <div
      className="content"
    >
      <div>
        <button
          className="lead "
          onClick={[MockFunction]}
        >
          title
        </button>
      </div>
      <div>
        Foo
      </div>
    </div>
  </div>
</div>
`;

exports[`Match snapshot with button 1`] = `
<div
  className="container"
>
  <div
    className="tile"
  >
    <div
      className="cover"
    >
      <img
        alt="Tile graphic"
        src="image"
      />
    </div>
    <div
      className="content"
    >
      <div>
        <button
          className="lead "
          onClick={[MockFunction]}
        >
          title
        </button>
      </div>
      <div>
        Foo
      </div>
      <button
        className="primary button  "
        onClick={[MockFunction]}
      >
        buttonText
      </button>
    </div>
  </div>
</div>
`;
